pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -95,38 +107,87 @@ def __init__(self, time, pqs):
         if self._pqs.shape[1] != 7:
             raise ValueError("`pqs` matrix shall have 7 columns.")
 
+        self._min_time = min(self.time)
```

Comment:
is there a reason you used the python inbuilts? or could we switch to np.min and np.max respectivly to have better performance?

Reply:
same further below with the 
```python
elif any(out_of_range):
```

could be changed to a
```python
elif np.any(out_of_range):
```

Reply:
In my experience the python builtins are faster as they have less overhead. Although we would have to test this for relevant array sizes.

Reply:
What are the relevant array sizes?
- lower end, just one transformation at one time (array size 1)
- upper end, Lidar-Pointcloud with time field, tranformation for every point (50_000 Points in a cloud)

i made a small microbenchmark with:
- AMD Ryzen 7 3700X
- Python 3.11.10
- numpy 2.1.3
- tested array sizes:  [5, 10, 50, 100, 500, 1000, 10_000, 100_000]

Verdict:
- inbuild any is always faster then numpy.any
- max/min , numpy is faster for arrays larger then `50`
- for an array of `10000`numpy is `150`-times faster

as the penalty of using numpy for small arrays is only minor i would suggest using `np.min/np.max` instead of the inbuilt ones and leave the inbuilt any. I know that these microbenchmarks can always be kinda missleading/error prone. But i think that the factor 150 for `10000`  is significant enough to make this decision. I attached the benchmark code as well

 
[micro_benchmark.py.txt](https://github.com/user-attachments/files/17871756/micro_benchmark.py.txt)

Reply:
Thanks for benchmarking. However, I would expect the overhead of using builtin min/max to be negligible when we have to compute ScLERP for more than 50 transformations. Could you test that? If that is the case, I think it makes more sense to optimize for small numbers of query times as I expect the case of querying 1 or a few times to be much more common.